packages/frontend/src/components/Listings.tsx

```diff
@@ -4,52 +4,67 @@
 
 import { useEffect, useState } from "react";
 
+import { Listing } from "@massmarket/schema";
+
 import CustomerViewListings from "./CustomerViewListings.tsx";
 import { useKeycard } from "../hooks/useKeycard.ts";
 import { useClientWithStateManager } from "../hooks/useClientWithStateManager.ts";
 import MerchantViewListings from "./merchants/listings/MerchantViewListings.tsx";
-import { asyncIteratorToMap } from "../utils/mod.ts";
+import { ListingId } from "../types.ts";
 
 export default function Listings() {
   const { clientStateManager, result } = useClientWithStateManager();
   const [keycard] = useKeycard();
-  const [products, setProducts] = useState(new Map());
+  const [products, setProducts] = useState<Map<ListingId, Listing>>(new Map());
+
+  function getAllListings() {
+    return clientStateManager!.stateManager.get(["Listings"]).then(
+      async (allListings: Map<ListingId, unknown>) => {
+        const listings = new Map();
+        for await (
+          const [
+            id,
+            o,
+          ] of allListings.entries()
+        ) {
+          listings.set(id, new Listing(o));
```

Comment:
see [93dd729](https://github.com/masslbs/Tennessine/pull/273/commits/93dd729e3608aba9c5926a3ac32a0f2b724479c5)

```suggestion
          listings.set(id, Listing.fromCBOR(o));
```